### PR TITLE
Add shared password input in network creation form.

### LINKED ISSUE #237: Add root password setup at LimeApp
We propose to allow network admin tasks such as firmware upgrade from the LimeApp.
The end goal is to help a network admin to upgrade the firmware in the entire mesh network from and assisted by the LimeApp.

For this goal to be achieved the very first step is to add the functionality for managing a shared-password at libremesh nodes.

The solution design proposal is:
- The password is setup at FirstBootWizard when creating a new network. And it's hash is saved at lime-community.
- A lime-config hook notice the hash at lime-community and setup the password for root user.

This issue asks for the need of the password input at FirstBootWizard in this context.

### PR DESCRIPTION
**Pull request by germanferrero**
This PR adds the input at FBW for setting a up the shared-password described at #237.

The password is sent to ubus endpoint lime-fbw -> create_network along with the network and host names.

closes #237

### CODE REVIEW

File: plugins/lime-plugin-fbw/src/containers/NetworkForm.js [hunk 1]
```diff
@@ -12,48 +12,72 @@ import I18n from 'i18n-js';
...
+		createNetwork({
+			network: state.communityName,
+			hostname: state.hostName,
+			password: state.password,
+		});
 		toggleForm('setting')();
 	}
 
+	function _isValidPassword() {
+		return state.password.length > 0;
```
Comment (spiccinini) [inline plugins/lime-plugin-fbw/src/containers/NetworkForm.js]: password length should be at least 10 to be valid, and with at least numbers and letters.

File: plugins/lime-plugin-fbw/src/containers/NetworkForm.js [hunk 2]
```diff
@@ -12,48 +12,72 @@ import I18n from 'i18n-js';
...
+		<label>{I18n.t('Re-enter the shared password')}</label>
+		<input type="password" placeholder={I18n.t('Re-enter Password')} class="u-full-width" value={state.passwordConfirmation} onInput={_changePasswordConfirmation} />
+		{_passwordMissMatch() &&
+			<label>{I18n.t('The passwords do not match!')}</label>
+		}
 		<label>{I18n.t('Choose a name for this node')}</label>
-		<input type="text" placeholder={I18n.t('Host name')} class="u-full-width" value={state.hostName} onChange={_changeHostName} onInput={_changeHostName} />
-		{/* <label>{I18n.t('Choose a password for this node')}</label>
-		<input type="text" placeholder={I18n.t('Password')} class="u-full-width" value={state.password} onChange={_changePassword} /> */}
+		<input type="text" placeholder={I18n.t('Host name')} class="u-full-width" value={state.hostName} onInput={_changeHostName} />
```
Comment (spiccinini) [inline plugins/lime-plugin-fbw/src/containers/NetworkForm.js:117]: I am not sure from the code, network, and hostname and password all required form values?

**Comment by spiccinini** [comment]
As we may implement also an option to select a WIFI password, if it is clearer if we rename this to `Admin password` or we can rename in the future when (and if) we add this functionality here.

File: plugins/lime-plugin-fbw/src/containers/NetworkForm.js (hunk 1 above)
Comment (germanferrero) [reply plugins/lime-plugin-fbw/src/containers/NetworkForm.js]: password validation is now implemented

File: plugins/lime-plugin-fbw/src/containers/NetworkForm.js (hunk 2 above)
Comment (germanferrero) [reply plugins/lime-plugin-fbw/src/containers/NetworkForm.js:117]: it was neither clear nor correct, thank you! Check it now :)

**Comment by spiccinini** [comment]
Tested! Great work :)

**Pull request merged**